enhancements/proposals/1978-scylladbdatacenter-api/README.md

```diff
@@ -0,0 +1,724 @@
+  Similarly, references to Scylla Manager Agent are renamed to ScyllaDB Manager Agent, even though the naming change isn't completed.
+- Instead of having a single field specifying the container image of used components,
+  there are two fields, one for the repository and the second for the version, which prevents from using 
+  full format of container images [#821](https://github.com/scylladb/scylla-operator/issues/821).
+  This unnecessary split is error-prone.
+- Kubernetes API Convention suggests using arrays instead of maps. The status of racks doesn't follow that rule.
+- `cpuSet` is confusing, as it doesn't change anything related to cpuset, it only controls stricter validation of allocated resources.
+  This is unnecessary when there is documentation about performance tuning containing these recommendations.
+  This field will be removed and validation will be relaxed.
+- We initially thought that `hostNetworking` helps with performance, but later we discovered we were wrong.
+  As we only see the downsides of using it, `hostNetworking` support will be discontinued, and setting it becomes deprecated
+  and discouraged by not having a field to set it in API. [#1378](https://github.com/scylladb/scylla-operator/issues/1378)
+  To preserve backward compatibility controllers will support enabling it by setting special 
+  annotation `scylla-operator.scylladb.com/enable-host-networking` to `true` on `v1alpha1.ScyllaDBDatacenter`.
+  The controller migrating from `v1.ScyllaCluster` to `v1alpha1.ScyllaDBDatacenter` will set this annotation when required.
+
+### Helm chart
+
+The existing `scylla` helm chart will use the existing `v1.ScyllaCluster`. It has several differences between what's in CRD
+and what's in the chart templates, hence it cannot be simply mapped one-to-one.
+A new helm chart called `scylladb-datacenter` will simply copy `spec` from `values` to a new CRD solving this issue and making it
+easily extensible, while keeping the schema of CRD and helm schema in sync.
+
+### Scylla Manager integration
+
+Scylla Manager tasks do not fit into a single datacenter, as Manager tasks work on the entire cluster.
+These are not yet transferred from the spec and status of the old API until a better alternative for these is available.
+The existing controller reconciling these will continue using `v1.ScyllaCluster`, users wanting to use these,
+won't be able to transition to the new API until an extension is available.
+Extending the manager integration may be addressed in a future proposal that rethinks the overall approach in more detail.
+
+### Topology placement
+
+Currently, to provide placement rules for a rack, users have to provide a very lengthy `nodeAffinity`.
+Even the simplest rules require providing 10'ish lines with 4'ish indentation levels.
+This makes the entire resource huge and UX isn't great.
+To improve it, the new field `toplogyLabelSelector` added on the datacenter and rack level,
```

Comment:
I planned to have two fields for topology label selector, one on DC level and second on rack level. Having a prefix for both of these might be confusing, as they are going to be merged together which is mentioned below.

Reply:
This also depends under which key this ends up being, https://github.com/scylladb/scylla-operator/pull/1994#discussion_r1664573792 so let's defer this

Reply:
Since it has different behavior than everything specified in rackTemplate (it's not overwriten but merged), I propose to keep it separate from the template.